client/pages/home.jsx

```diff
@@ -1,31 +1,82 @@
 import React from 'react';
 import ChatListSection from '../components/chat-list-section';
 import MessageArea from '../components/message-area';
-import parseRoute from '../lib/parse-route';
+import CreateUserForm from '../components/create-user-form';
+import { parseRoute, decodeToken, parseChatRooms } from '../lib';
 
 export default class Home extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      route: parseRoute(window.location.hash)
+      route: parseRoute(window.location.hash),
+      user: null
     };
+    this.submitNewUser = this.submitNewUser.bind(this);
+    this.addRoom = this.addRoom.bind(this);
   }
 
   componentDidMount() {
     window.addEventListener('hashchange', event => {
       const parsedHash = parseRoute(window.location.hash);
-      this.setState({ route: parsedHash });
+      this.setState(state => {
+        return ({
+          route: parsedHash,
+          user: state.user
+        });
+      });
     });
+    const token = window.localStorage.getItem('chat-app-jwt');
+    let user = token ? decodeToken(token) : null;
+    if (user) {
+      user = parseChatRooms(user);
+    }
```

Comment:
You can leave this part out if you switch the chatRooms column to json instead of text.